pytorch_lightning/callbacks/lr_monitor.py

```diff
@@ -168,35 +183,45 @@ def _extract_momentum(self, param_group, name: str, use_betas: bool) -> Dict[str
         self.last_momentum_values[name] = momentum
         return {name: momentum}
 
-    def _find_names(self, lr_schedulers) -> List[str]:
-        # Create uniqe names in the case we have multiple of the same learning
-        # rate schduler + multiple parameter groups
+    def _add_prefix(self, name, optimizer_cls, seen_optimizer_types) -> str:
+        count = seen_optimizer_types[optimizer_cls]
+        return name + f'-{count}' if count else name
+
+    def _find_names(self, lr_schedulers, add_lr_sch_names: bool = True) -> List[str]:
+        # Create unique names in the case we have multiple of the same learning
+        # rate scheduler + multiple parameter groups
         names = []
+        seen_optimizers = []
+        seen_optimizer_types = defaultdict(int)
         for scheduler in lr_schedulers:
             sch = scheduler['scheduler']
             if scheduler['name'] is not None:
                 name = scheduler['name']
             else:
-                opt_name = 'lr-' + sch.optimizer.__class__.__name__
-                i, name = 1, opt_name
+                name = 'lr-' + sch.optimizer.__class__.__name__
 
-                # Multiple schduler of the same type
-                while True:
-                    if name not in names:
-                        break
-                    i, name = i + 1, f'{opt_name}-{i}'
+            seen_optimizers.append(sch.optimizer)
+            optimizer_cls = type(sch.optimizer)
+            if scheduler['name'] is None:
+                if optimizer_cls not in seen_optimizer_types:
```

Comment:
As `seen_optimizer_types` is a `defaultdict` you should just be able to do `seen_optimizer_types[optimizer_cls] += 1` and not need the if.

Reply:
Done !

Reply:
Doesn't look done 🧐

Reply:
Added back !